dot/state/tries_test.go

```diff
@@ -49,7 +49,7 @@ func Test_Tries_SetEmptyTrie(t *testing.T) {
 func Test_Tries_SetTrie(t *testing.T) {
 	t.Parallel()
 
-	tr := trie.NewTrie(&node.Node{PartialKey: []byte{1}})
+	tr := trie.NewTrie(&node.Node{PartialKey: []byte{1}}, nil)
```

Comment:
since this is done a lot, maybe just create a helper function in tests that uses a mock `db.Database` for you.